atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/OracleDbTableFactory.java

```diff
@@ -83,31 +73,30 @@ public DbReadTable createRead(String tableName, ConnectionSupplier conns) {
     }
 
     @Override
-    public DbWriteTable createWrite(String tableName, ConnectionSupplier conns) {
-        TableSize tableSize = getTableSize(conns, tableName);
+    public DbWriteTable createWrite(TableReference tableRef, ConnectionSupplier conns) {
+        TableSize tableSize = getTableSize(conns, tableRef);
         switch (tableSize) {
             case OVERFLOW:
-                return new OracleOverflowWriteTable(tableName, conns, config);
+                return OracleOverflowWriteTable.create(DbKvs.internalTableName(tableRef), conns, config);
             case RAW:
-                return new SimpleDbWriteTable(tableName, conns, config);
+                return new SimpleDbWriteTable(DbKvs.internalTableName(tableRef), conns, config);
             default:
                 throw new EnumConstantNotPresentException(TableSize.class, tableSize.name());
         }
     }
 
-    private TableSize getTableSize(final ConnectionSupplier conns, final String tableName) {
+    private TableSize getTableSize(final ConnectionSupplier conns, final TableReference tableRef) {
         try {
-            return tableSizeByTableName.get(tableName, new Callable<TableSize>() {
+            return tableSizeByTableName.get(tableRef.getQualifiedName(), new Callable<TableSize>() {
                 @Override
                 public TableSize call() {
-                    String metadataTableName = config.metadataTable().getQualifiedName();
                     AgnosticResultSet results = conns.get().selectResultSetUnregisteredQuery(
-                            String.format("SELECT table_size FROM %s WHERE table_name = ?", metadataTableName),
-                            tableName);
+                            String.format("SELECT table_size FROM %s WHERE table_name = ?", config.metadataTableName()),
+                            tableRef.getQualifiedName());
```

Comment:
It worked fine in an internal product as ``internalTableName`` and ``qualifiedName`` were the same.